enhancements/cluster-logging/gcp-external-account-auth.md

```diff
@@ -0,0 +1,201 @@
+* Ensure compatibility with existing service account authentication
+
+
+### Non-Goals
+
+* Modifying the OpenShift gcp output authentication model. 
+* Supporting authentication methods outside of Google Cloud.
+
+
+## Proposal
+
+To integrate WIF, Vector’s authentication logic will be updated to support external_account 
+credentials in addition to the existing service_account credentials. The process will:
+
+* Determine the authentication type by reading the credentials JSON file. 
+* Extract the OpenShift Service Account token from the projected volume. 
+* Exchange the OpenShift token for a Google Identity Token using Google STS. 
+* Exchange the Identity Token for a short-lived Access Token via Google IAM impersonation. 
+* Ensure the existing service account authentication remains functional.
+
+### Workflow Description
+### API Extensions
+The existing credentials key and secret works for our purpose, although we should aim to eventually  
+align with our cloudwatch and future Azure specs which are most likely going to be configMaps:
+```yaml
+ googleCloudLogging:
+   authentication:
+     credentials:
+       key: google-ext-account-creds.json
+       secretName: gcp-secret
+   logId : my-gcp
+...
+```
+### Topology Considerations
+#### Hypershift / Hosted Control Planes
+#### Standalone Clusters
+#### Single-node Deployments or MicroShift
+### Implementation Details/Notes/Constraints
+
+1. Modify Authentication to Support external_account  
+   * Update authentication logic to check for external_account type.
+   * Add a CredentialsType struct to differentiate between service_account and external_account.
+2. Introduce ExternalCredentials Struct   
+   * Create a new struct to handle external_account authentication fields.
+   * Ensure compatibility with Google’s OIDC-based credential format.
+3. Implement the WIF Authentication Flow   
+   * Step 1: Read the OpenShift service account token from the path specified in the credentials file   
+   * Step 2: Exchange the OpenShift token for a Google Identity Token via STS.   
+   * Step 3: Exchange the Identity Token for an Access Token via IAM impersonation.  
+4. Ensure Compatibility with service_account Type  
+   * The existing authentication flow for service_account remains unchanged.
+
+#### Credentials File Example
+(external_account type)
+```json
+{
+  "type": "external_account",
+  "audience": "//iam.googleapis.com/...<workload_identity_pool_provider>",
+  "token_url": "https://sts.googleapis.com/v1/token",
+  "service_account_impersonation_url": "https://iamcredentials......iam.gserviceaccount.com:generateAccessToken",
+  "credential_source": { 
+    "file": "/var/run/ocp-collector/serviceaccount/token"
+  }
+}
+
+```
+
+#### OIDC Request Examples
+
+##### How Is The Bound Token Used in GCP Workload Identity Federation (WIF)?
+1. Pod gets the Bound Service Account Token
+   * This token is created automatically by OpenShift and mapped to the service account assigned to the pod.
```

Comment:
Is this the projected token?  My understanding was that auto SA token generated by simply creating an SA is no longer supported.  @xperimental do you know?

Reply:
This is the projected service account token, also called a "bound token" in WIF parlance, and is required as part of the OIDC process.    Any SA token could be used, if it was annotated correctly, but this is out of scope for our feature.   We want to use our projected token.     
  I've updated the verbiage to be more clear.